romancal/photom/photom.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import warnings
+
+import numpy as np
+from astropy import units as u
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+# Conversion factor from MJy/sr to uJy/arcsec^2
+MJSR_TO_UJA2 = (u.megajansky / u.steradian).to(u.microjansky / u.arcsecond / u.arcsecond)
+
+# Conversion factor from square arcseconds to steradians
+A2_TO_SR = (np.pi / (180. * 3600.))**2
+
+
+def photom_io(input_model, photom_metadata):
+    """
+    Short Summary
+    -------------
+    Combine photometric scalar conversion factors and add to the science metadata.
+
+    Parameters
+    ----------
+    input_model : Roman level 2 image datamodel
+        input roman datamodel
+
+    photom_metadata : list
+        Set of matched photom meta data keys
+
+    Returns
+    -------
+
+    """
+    # Get the scalar conversion factor.
+    conversion = photom_metadata['photmjsr']  # unit is MJy / sr
+
+    # Store the conversion factor in the meta data
+    log.info(f'photmjsr value: {conversion:.6g}')
+    input_model.meta.photometry.conversion_megajanskys = conversion
+    input_model.meta.photometry.conversion_microjanskys = conversion.value * MJSR_TO_UJA2 *\
+                                                          u.microjansky / (u.arcsecond**2)
```

Comment:
Why not use units directly, instead of going through `.value`?
For example, `conversion.to(u.microjansky / u.arcsecond**2)`

Reply:
You caught a bug in RDM! This wasn't working, hence the conversion factors, but with your comment, I dug into it more deeply. It turns out that the unit was set to u.MJ instead of u.MJy, which yielded conversion errors. I will make a ticket for the RDM correction, and correct the code in RCAL.

Reply:
The RDM fixes are in:
https://github.com/spacetelescope/roman_datamodels/pull/73